### ISSUE TITLE
PAL Color Monitor support

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#763]
@Archange427 emailed me on 17/2/2020:

---

I noticed that with AppleWin's NTSC video mode, we can see color "artifacts" around letters when we mix HIRES (GR too?) mode and Text mode like for example in "Unreeeal Superhero Tribute"  (see picture above:)

![screen1](https://user-images.githubusercontent.com/6696896/75115122-bd836280-5653-11ea-8c52-f117ee5df0a5.jpg)
=> Magenta and green "artifacts" around letters...

I don't notice this kind of artifacts with PAL display on my APPLE II with Color Monitor (composite output)
Even when using a mixed custom mode, the text is as sharper as if we were using normal text mode.

I think this is obviously the normal behavior of the NTSC mode perfectly emulated by AppleWin.
But now that we have a 50 Hz PAL mode (I can't thank you enough for that!!!), maybe we can also have PAL display mode? :)

I think there are probably other differences between NTSC and PAL display, even with the normal gfx modes, the "famous" NTSC artifacts are probably not present.

**Comment by tomcw** [comment]
So should the `50Hz video` checkbox also render the TEXT without the colour fringing, for 'Color (NTSC Monitor)'... effectively making it 'Color (PAL Monitor)'?

btw. 'Color (RGB Monitor)' already renders the TEXT without colour fringing:
![tribu-rgb-monitor](https://user-images.githubusercontent.com/6696896/75115400-25d34380-5656-11ea-95b0-38799d7fc1e3.png)

**Comment by Archange427** [comment]
In this (very good) post - https://github.com/AppleWin/AppleWin/issues/764#issuecomment-590133459 - @fenarinarsa said that the colors in PAL are not exactly the same as in NTSC.
So the TEXT without the color fringing does not seem to be the only difference.

**Comment by fenarinarsa** [comment]
Indeed. In "Understading the Apple IIe" there's a mention that the chromaburst is not disabled on the text part in NTSC. But it's not explained how or why so this info may be bogus... actually the rest of the video generation chapters seems to say exactly the opposite (that chromaburst IS indeed disabled on the text part). But I got at least two people telling me that's actually the behavior of NTSC monitors that can't disable chroma decoding during a frame. That means even if the Apple II does disable chromaburst, an NTSC monitor will continue to handle chroma anyway until the next frame starts.

The reason why I say the info might be bogus is that the PAL video generator isn't a NTSC>PAL converter, but a complete modulator that take the NTSC chroma as well as all the various softswitch signals to create the PAL composite video. And if the NTSC chromaburst is disabled, the PAL modulator doesn't add chroma to the current line. And that's what we observe in PAL mixed mode.

**Comment by tomcw** [comment]
>that's actually the behavior of NTSC monitors that can't disable chroma decoding during a frame.

On 01/10/2018, MJ Mahon comments on this here:
https://groups.google.com/d/msg/comp.sys.apple2/gNa-899NL7E/poUOZW8NAgAJ

>Most analog monitors will take many lines of ~~burst it~~ no burst to switch
between color and monochrome display modes. In NTSC television broadcasts
(RIP), the burst switching usually occurred both preceded and followed by
several black frames, so the transition time was not critical and invisible
to viewers.
>
>Fast switching was a bad idea, because interference could frequently result
in several lines of corrupted color bursts, and sets were designed to “ride
through” these interruptions. The color burst served to phase lock a
crystal oscillator which could “flywheel” through several lines of missing
burst with little phase drift.
>
>Digital monitors may or may not respond quicker, but if they were designed
for off-the-air reception, they should have made similar provisions for
noise immunity.

**Comment by sicklittlemonkey** [comment]
> In "Understading the Apple IIe" there's a mention that the chromaburst is not disabled on the text part in NTSC. But it's not explained how or why so this info may be bogus... actually the rest of the video generation chapters seems to say exactly the opposite (that chromaburst IS indeed disabled on the text part). But I got at least two people telling me that's actually the behavior of NTSC monitors that can't disable chroma decoding during a frame. That means even if the Apple II does disable chromaburst, an NTSC monitor will continue to handle chroma anyway until the next frame starts.

Sather made very few mistakes in his books, but this might (just might) be one of them. The schematic in Figure 8.5 has CLRGATE' derived from the IOU TEXT softswitch signal, which he states explicitly. The mention in question is Understanding the Apple //e page 8-16, 1st column:

> COLOR BURST' is a 14-cyc!e (four video scanner
> states) inverted sample of the COLOR REFERENCE
> signal. It is gated by CLRGATE' from the
> lOU which drops low immediately after horizontal
> sync if the TEXT softswitch is reset (see Figure 8.2).
> CLRGATE' never drops low ifTEXT is set, so there
> is no COLOR BURST, and consequently no undesirable
> coloring of screen characters, in TEXT mode.
> ColoringofTEXTcharactera does occur in MIXED
> GRAPHICS mode because the COLOR BURST is
> not turned off, even during TEXT time.

> The reason why I say the info might be bogus is that the PAL video generator isn't a NTSC>PAL converter, but a complete modulator that take the NTSC chroma as well as all the various softswitch signals to create the PAL composite video. And if the NTSC chromaburst is disabled, the PAL modulator doesn't add chroma to the current line. And that's what we observe in PAL mixed mode.

The PAL white mixed text is the only contradiction. I don't have a scope but it should be pretty easy for someone who has one to confirm or refute this.

The behaviour of color burst persisting for several frames has been stated multiple times by Michael Mahon - who is almost as authoritative a source as Sather - so Tom is quite right to quote him. ; - )

As for Tom's question:

> So should the 50Hz video checkbox also render the TEXT without the colour fringing, for 'Color (NTSC Monitor)'... effectively making it 'Color (PAL Monitor)'?

I think the answer is yes.

Color (RGB Monitor) has no color "artifacts" at all. Color (PAL Monitor) would be the same as Color (NTSC Monitor), i.e with the HGR colour artifacts, but not the text colour artifacts.

According to the schematic, PAL colour is basically NTSC color but sampled into 16 (4 bits) discrete colours represented by the resistor summing network.

Tom, do you have a II Eurocolor card to compare with?

**Comment by fenarinarsa** [comment]
> 
> The PAL white mixed text is the only contradiction. I don't have a scope but it should be pretty easy for someone who has one to confirm or refute this.

Chapter 8-17 is about PAL video generation. Sather says the PAL modulator is the same than on Eurocolor card.

I had feedbacks from PAL Apple IIe users with composite output that told me there is no visible color fringe on the text in mixed mode.

**Comment by sicklittlemonkey** [comment]
PAL composite mixed mode text definitely has no colour.

It's just not clear from Sather how it's implemented.

**Comment by tomcw** [comment]
>Tom, do you have a II Eurocolor card to compare with?

Sadly not - all I have (for colour) is a PAL CRT TV and a TV modulator.

**Comment by tomcw** [comment]
@Archange427 sent me a patch for this.
For now, I've create a new branch (GH763-PAL-TEXT) and commit his patch to that branch:
https://github.com/AppleWin/AppleWin/commits/GH763-PAL-TEXT

One thing I noticed is that the TEXT40 and TEXT80 (in MIXED mode) needs to be shifted by 1 half-pixel (14MHz dot) for correct alignment with the graphics.
EG, when running test E & F from:
https://github.com/AppleWin/AppleWin-Test/blob/master/v2/Tests-Various.dsk

**Comment by sicklittlemonkey** [comment]
I wouldn't worry too much about one pixel. Look at what happens on a real Apple (//c) when the text is 80 but DHIRES is off ...
![LeftEsc80](https://user-images.githubusercontent.com/4393005/89546204-af16b500-d847-11ea-92ce-b162e8f38e47.jpg)

**Comment by tomcw** [comment]
AppleWin's title bar also changes to reflect this: "Color (NTSC Monitor)" -> "Color (PAL Monitor)".
(I considered spelling this "Colour (PAL Monitor)", but only after making the change.)

**Comment by tomcw** [comment]
>One thing I noticed is that the TEXT40 and TEXT80 (in MIXED mode) needs to be shifted by 1 half-pixel (14MHz dot) for correct alignment with the graphics.

...and I fixed the alignment issues.

**Comment by tomcw** [comment]
@Archange427 - there's a new AppleWin 1.29.14.0 [here](https://github.com/AppleWin/AppleWin/releases/download/v1.29.14.0/AppleWin1.29.14.0.zip), which should fix this issue. Can you give it a try?

**Comment by Archange427** [comment]
Works fine!
Thanks for the update

Tested with Color TV too => title bar is the same even if 50Hz is checked. 
Shouldn't it be "Color TV (PAL)" ?

**Comment by tomcw** [comment]
>Shouldn't it be "Color TV (PAL)" ?

re. app window's title:
- Currently it says "Color TV"... so it'd mean **dynamically** renaming it to either ""Color TV (NTSC)" or "Color TV (PAL)".
- And "B&W TV" should similarly be either ""B&W TV (NTSC)" or "B&W TV (PAL)".
- NB. "Green/Amber/White Monitor" can all remain as-is.

I also need to update the Configuration's drop-down menu:
- "Color (NTSC Monitor)" -> "Color (Composite Monitor)"
- NB. "Color TV" and "B&W TV" can both remain as-is.

**Comment by Archange427** [comment]
is there a difference between B&W NTSC and PAL rendering?

**Comment by tomcw** [comment]
The only difference is 262 vs 312 lines... so it may be useful to know (from the app window's title bar) that it's in NTSC or PAL mode.

**Comment by Archange427** [comment]
So, as long as the 50Hz box is ticked, you should specify PAL regardless of the mode (RGB, COLOR TV, MONOCHROME,...)
